client/src/hoc/with-search/index.js

```diff
@@ -0,0 +1,153 @@
+  const displayName = `HigherOrder.WithSearchWrapper('${getDisplayName(
+    WrappedComponent
+  )})`;
+
+  class WithSearch extends PureComponent {
+    static WrappedComponent = WrappedComponent;
+    static displayName = displayName;
+
+    static mapStateToProps = state => {
+      return {
+        results: select(requests.rSearchResults, state.entityStore),
+        resultsMeta: meta(requests.rSearchResults, state.entityStore),
+        searchQueryState: get(state.ui, "transitory.search")
+      };
+    };
+
+    static propTypes = {
+      dispatch: PropTypes.func.isRequired,
+      history: PropTypes.object.isRequired,
+      location: PropTypes.object.isRequired,
+      searchQueryState: PropTypes.object.isRequired
+    };
+
+    constructor(props) {
+      super(props);
+      this.searchComponent = React.createRef();
+    }
+
+    // If the search container mounts with searchQueryState params, it's being opened because
+    // a search has been made already.  Immediately update the store, which will cause a search request.
+    componentDidMount() {
+      const locationSearchStateQuery = get(
+        this.props.location,
+        "state.searchQueryState"
+      );
+      if (locationSearchStateQuery)
+        return this.setQueryState(locationSearchStateQuery);
+    }
+
+    componentDidUpdate(prevProps, prevStateIgnored) {
+      // If the search query changes in the store, we know that new params were entered
+      // and we have to make a new search request.
+      if (prevProps.searchQueryState !== this.props.searchQueryState) {
+        return this.doSearch();
+      }
+
+      // If we get a new searchQueryState as location state, we update the search query in the store.
+      // This comes back as our searchQueryState prop, which will kick off a new search if changed.
+      // The location state is reset after a search is performed (see doSearch promise success callback)
+      // to ensure the location state has actually been changed.
+      const locationSearchStateQuery = get(
+        this.props.location,
+        "state.searchQueryState"
+      );
+      const prevLocationSearchStateQuery = get(
+        prevProps.location,
+        "state.searchQueryState"
+      );
+      if (!locationSearchStateQuery) return null;
+
+      if (prevLocationSearchStateQuery !== locationSearchStateQuery) {
+        return this.setQueryState(locationSearchStateQuery);
+      }
+    }
+
+    componentWillUnmount() {
+      this.props.dispatch(flush(requests.rSearchResults));
+      this.props.dispatch(uiSearchActions.resetSearch());
+    }
+
+    get childProps() {
+      return {
+        setQueryState: this.setQueryState,
+        setPage: this.setPage,
+        searchStateQuery: this.props.searchQueryState,
+        history: this.props.history,
+        location: this.props.location,
+        ref: this.searchComponent
+      };
+    }
+
+    /* eslint-disable no-console */
+    get searchParams() {
```

Comment:
This is the only part I'm not excited about.  It works, and should continue to work, but it would be nice if there was a better way to define the logic for facets in a search container than calling the child component method.  Static methods would be slightly better, but we still need access to props in them.